lib/hash.js

```diff
@@ -0,0 +1,74 @@
+const crypto = require('crypto');
+
+const SALT_LENGTH = 64;
+const HASH_ITERATIONS = 10000;
+const HASH_LENGTH = 256;
+const CIPHER = 'sha512';
+const SEPARATOR = ':';
+
+function genSalt() {
+  return crypto.randomBytes(SALT_LENGTH).toString('hex');
+}
+
+/**
+ * @callback saltAndHashCallback
+ * @param {Error}
+ * @param {String} responseMessage
+ */
+
+/**
+ * Generates a random salt and encrypts the given password
+ * @param  {String}   pwd Password to be encrypted
+ * @param  {saltAndHashCallback} cb Node-style callback
+ */
+exports.saltAndHash = function(pwd, cb) {
+  const salt = genSalt();
+  crypto.pbkdf2(pwd, salt, HASH_ITERATIONS, HASH_LENGTH, CIPHER, function(err, hashed) {
```

Comment:
@ziccardi Thanks for the input, I was thinking about this earlier and implementing an exponential delay at the store level.

@nialldonnellyfh Do you think this is the way to go or should we move this to another issue to decide on password policy with related UI features? (i.e. automatic locking instead)
I think we also need support for password change tokens, or at least just allowing someone with access to the portal to reset an user's password to a random one.